flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -293,13 +293,25 @@ abstract class TableEnvironment(val config: TableConfig) {
     }
   }
 
+  private def preOptimizeLogicalPlan(relNode: RelNode): RelNode = {
```

Comment:
why is the normalization rule set not enough?

Reply:
good idea!